registrations/tasks.py

```
@@ -390,3 +393,149 @@ def run(
+
+                reg_info = {
+                    "stage": line['type_of_registration'],
+                    "mother_id": mother_id,
+                    "source": source.id,
+                    "data": {
+                        "msg_receiver":
+                            self.get_receiver(line['message_receiver']),
+                        "operator_id": operator_id,
+                        "language":
+                            self.get_language(line['preferred_msg_language']),
+                        "msg_type": msg_type
+                    }
+                }
+
+                if line['gatekeeper_phone_number']:
+                    receiver_id = self.get_or_create_identity(
+                        line['gatekeeper_phone_number'])
+                    reg_info['data']['receiver_id'] = receiver_id
+
+                if msg_type == 'audio':
+                    reg_info['data']['voice_times'] = self.get_voice_times(
+                        line['message_time'])
+                    reg_info['data']['voice_days'] = self.get_voice_days(
+                        line['message_days'])
+
+                if line['type_of_registration'] == 'prebirth':
+                    reg_info['data']['last_period_date'] = \
+                        utils.calc_date_from_pregnancy_week(
+                            utils.get_today(),
+                            line["pregnancy_week"]).strftime("%Y%m%d")
+                else:
+                    reg_info['data']['baby_dob'] = utils.calc_baby_dob(
+                        utils.get_today(),
+                        line["pregnancy_week"]).strftime("%Y%m%d")
+
+                serializer = RegistrationSerializer(data=reg_info)
+                serializer.is_valid(raise_exception=True)
+
+                Registration.objects.create(**serializer.validated_data)
+            except Exception, error:
+                line['error'] = str(error)
+                ThirdPartyRegistrationError.objects.create(**{'data': line})
```

I think maybe we should do something like have a variable outside of the loop that keeps the value of the last error, or None if there are no errors. If that variable contains an error after the loop, then we should raise that error, else we do nothing.

That way, if there's an error on any of the rows, we get a Sentry notification about it.